File: app/src/main/java/ru/arhlib/app/afisha/Page.java
package ru.arhlib.app.afisha;

import com.google.gson.annotations.SerializedName;

import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity
public class Page {
    @PrimaryKey
    public Integer id;
    @SerializedName("content")
    @Embedded
    public Rendered content;

    public Page(Integer id, Rendered content) {
        this.id = id;
        this.content = content;
    }

    public String getContent() {
        return content.rendered;
    }

    static class Rendered {
        String rendered;
    }
}

File: app/src/main/java/ru/arhlib/app/afisha/Page.kt
package ru.arhlib.app.afisha

import kotlinx.serialization.Serializable

@Serializable
data class Page(
        val id: Int,
        private val title: Rendered,
        private val content: Rendered,
) {
    fun getRenderedTitle(): String {
        return title.rendered
    }

    fun getRenderedContent(): String {
        return content.rendered
    }

    @Serializable
    data class Rendered(
            val rendered: String,
    )
}
